### DISCUSSION TITLE
How to normalize the descriptors for deepmd?

### DISCUSSION THREAD

**Discussion by lhycms** [deepmodeling/deepmd-kit#2589] [Q&A] [answered]
It seem that we should use `davg` and `dstd` to normalize descriptors before training, and the inference process also requires these two data. But i can't figure out how to calculate `davg` and `dstd`. 
Are they the statistic data for `(s, sx/r, sy/r, sz/r)`? But the feature of deepmd is `G^T R R^T G`, if we just normalize `(s, sx/r, sy/r, s*z/r)`, we are unable to guarantee features normalized.
Looking forward to everyone joining the discussion!

**Comment by njzjz** [comment, marked as answer]
`davg` and `dstd ` is only used to normalize the environmental matrix $\mathcal{R}^i$.  $\mathcal{G}^i$ is derived from $\mathcal{R}^i$, so we expect it does not need to be normalized. $\mathcal{D}^i$ is given by
```math
\mathcal{D}^i = \frac{1}{N_c^2} (\mathcal{G}^i)^T \mathcal{R}^i (\mathcal{R}^i)^T \mathcal{G}^i_<
```

**Reply by lhycms** [reply]
Thanks for th help!
